common/chains/client/rpc.go

```diff
@@ -0,0 +1,105 @@
+package client
+
+import (
+	"context"
+	"math/big"
+
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+	"github.com/smartcontractkit/chainlink/v2/core/assets"
+)
+
+// RPC includes all the necessary methods for a multi-node client to interact directly with any RPC endpoint.
+type RPC[
+	CHAIN_ID types.ID,
+	SEQ types.Sequence,
+	ADDR types.Hashable,
+	BLOCK_HASH types.Hashable,
+	TX any,
+	TX_HASH types.Hashable,
+	EVENT any,
+	EVENT_OPS any,
+	TX_RECEIPT types.Receipt[TX_HASH, BLOCK_HASH],
+	FEE feetypes.Fee,
+	HEAD types.Head[BLOCK_HASH],
+
+] interface {
+	NodeClient[
+		CHAIN_ID,
+		HEAD,
+	]
+	ClientAPI[
+		CHAIN_ID,
+		SEQ,
+		ADDR,
+		BLOCK_HASH,
+		TX,
+		TX_HASH,
+		EVENT,
+		EVENT_OPS,
+		TX_RECEIPT,
+		FEE,
+		HEAD,
+	]
+}
+
+type ClientAPI[
```

Comment:
What else does `API` tell the reader compared to just `Client`?
```suggestion
type Client[
```
`Client` may not say enough, and I don't love having `client.Client`, but can we do better than `API`? Maybe adding a doc comment here would help shake out a better name.

Reply:
`Client` is the user-facing interface that has all the necessary methods called by a user that imports the common client. `ClientAPI` has all the RPC methods that are required by the user-facing `Client`. For example `BatchCallContextAll` is a method exported by that client, but there is no direct RPC call for that. You only need `BatchCallContext` on the underlying `ClientAPI` to implement it.
I also struggled to find a good name to describe this, that's why I added the comment down below.